openzim/kolibri
# New UI step 1

This is a reopen of #65 with an intermediate base branch `new_ui` since know limitation below prevent us to merge to`main`

Changes:
- Scraper (Python code) has been moved to the `scraper` subfolder
- Vue.JS is now used as main UI framework ; all its code is in the `zimui` subfolder ; it is rendered with Vite to produce a static website
- QA and Tests workflows have been adapted 
  - to the new folder structure 
  - to also QA and Test the Vue.JS part
- precommit hooks have been configured for the Vue.JS part
- Dockerfile has been adapted to first build the Vue.JS part in a dedicated stage and then embed the generated files into the final Python-based image
- Topics are stored as JSON files in the ZIM
  - JSON is generated by `pydantic`
  - these files are consumed by the Vue.JS UI
  - content (video, audio, pdf, epub, ...) is still rendered by Jinja2 as before
- URLs are meaningful slugs
  - generated by Python `slugify` lib
  - from Kolibri node title
  - should two distinct nodes have the same title resulting in the same slug, conflicts are handled with a `_1`, `_2`, ... suffix
- changes in the ZIM "folder" structure:
  - files generated by Vite are placed in `/`
  - thumbnails are placed in `/thumbnails`
  - JSON files generated to render topics are placed in `/topics`
  - most Kolibri content (video, audio, ePub, PDF) are placed in `/files` (some content is still placed at the root to not break some stuff which was found hard to fix for now, will be tackled in specific issues for each content type)
- legacy MANIFEST.in has been deleted (left-over from migration to hatch)
- `is_front` property has been adjusted when adding the item to the ZIM
- one new CLI argument `--zimui-dist` to specify the folder where zimui has been built (by Vite)

Known side effects to be discussed:
- topics are not searchable anymore (will be fixed by #67)
- navigation breadcrumbs are not here anymore (will be fixed by #64)
